db/batch/batch_impl.go

```diff
@@ -133,6 +136,17 @@ func (b *baseKVStoreBatch) Translate(wit WriteInfoTranslate) KVStoreBatch {
 	return c
 }
 
+func (b *baseKVStoreBatch) CheckFillPercent(ns string) (float64, bool) {
+	p, ok := b.fill[ns]
+	return p, ok
```

Comment:
why read doesn't need rlock?

Reply:
read is actually called in WriteBatch(), where the lock is obtained before writing all entries to DB
so cannot rlock here, otherwise it is deadlock

Reply:
in that case, i think it could be more clear to have a separate lock to lock different resources? It is very weird that `CheckFillPercent` and `AddFillPercent` access same resource, one has lock, one doesn't

Reply:
yeah, fixed in latest push